src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/NeighborPackagedData.hpp

```diff
@@ -0,0 +1,231 @@
+        using dg_package_data_argument_tags = tmpl::append<
+            evolved_vars_tags, recons_prim_tags, fluxes_tags,
+            tmpl::remove_duplicates<tmpl::push_back<
+                dg_package_data_temporary_tags, gr::Tags::SpatialMetric<3>,
+                gr::Tags::SqrtDetSpatialMetric<DataVector>,
+                gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataVector>,
+                evolution::dg::Actions::detail::NormalVector<3>>>>;
+
+        const auto& element = db::get<domain::Tags::Element<3>>(box);
+        const auto& eos = get<hydro::Tags::EquationOfStateBase>(box);
+
+        for (const auto& mortar_id : mortars_to_reconstruct_to) {
+          const Direction<3>& direction = mortar_id.first;
+
+          Index<3> extents = subcell_mesh.extents();
+          ++extents[direction.dimension()];
+
+          // Computed prims and cons on face via reconstruction
+          const size_t num_face_pts = subcell_mesh.extents()
+                                          .slice_away(direction.dimension())
+                                          .product();
+          Variables<dg_package_data_argument_tags> vars_on_face{num_face_pts};
+          // Copy spacetime vars over from volume.
+          using spacetime_vars_to_copy = tmpl::list<
+              gr::Tags::Lapse<DataVector>,
+              gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+              gr::Tags::SpatialMetric<3>,
+              gr::Tags::SqrtDetSpatialMetric<DataVector>,
+              gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataVector>>;
+          tmpl::for_each<spacetime_vars_to_copy>(
+              [&direction, &extents, &vars_on_face,
+               &spacetime_vars_on_faces =
+                   db::get<evolution::dg::subcell::Tags::OnSubcellFaces<
+                       typename System::flux_spacetime_variables_tag, 3>>(box)](
+                  auto tag_v) noexcept {
+                using tag = tmpl::type_from<decltype(tag_v)>;
+                data_on_slice(make_not_null(&get<tag>(vars_on_face)),
+                              get<tag>(gsl::at(spacetime_vars_on_faces,
+                                               direction.dimension())),
+                              extents, direction.dimension(),
+                              direction.side() == Side::Lower
+                                  ? 0
+                                  : extents[direction.dimension()] - 1);
+              });
+
+          call_with_dynamic_type<void, typename grmhd::ValenciaDivClean::fd::
+                                           Reconstructor::creatable_classes>(
+              &recons, [&element, &eos, &mortar_id, &nhbr_subcell_data,
+                        &subcell_mesh, &vars_on_face,
+                        &volume_prims](const auto& reconstructor) noexcept {
+                reconstructor->reconstruct_fd_neighbor(
+                    make_not_null(&vars_on_face), volume_prims, eos, element,
```

Comment:
For my understanding -- can you explain what this is doing for the `vars_on_face`? A naive reading would imagine the `vars_on_face` to now be on the dg grid, but then they are used in the `dg_package_data` below before the `fd::reconstruct`, so I must somehow be misunderstanding the control flow.

Reply:
Ah, I see, I think that does clarify matters for me.